src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -39,24 +39,24 @@ internal sealed partial class AssetProvider(Checksum solutionChecksum, SolutionA
         using var _1 = PooledHashSet<Checksum>.GetInstance(out var checksums);
         checksums.Add(checksum);
 
-        using var _2 = PooledDictionary<Checksum, object>.GetInstance(out var results);
+        using var _2 = PooledDictionary<Checksum, T>.GetInstance(out var results);
```

Comment:
the method this is in is already generic.  so this now allows that generic information to flow *into* SynchronizeAssetsAsync